mrp_production_auto_validate/models/mrp_production.py

```diff
@@ -12,36 +12,38 @@ class MrpProduction(models.Model):
     _inherit = "mrp.production"
 
     auto_validate = fields.Boolean(
-        string="Auto Validate",
         compute="_compute_auto_validate",
         store=True,
-        states={"draft": [("readonly", False)]},
     )
 
     @api.constrains("bom_id", "auto_validate", "product_qty")
     def check_bom_auto_validate(self):
+        bypass_check = self.env.context.get("disable_check_mo_auto_validate")
+        if bypass_check:
+            return
+
         for mo in self:
-            # FIXME: Handle different UOM between BOM and MO
-            qty_ok = (
-                tools.float_compare(
-                    mo.product_qty,
-                    mo.bom_id.product_qty,
-                    precision_rounding=mo.product_uom_id.rounding,
+            if mo.bom_id:
+                qty_in_mo_uom = mo.bom_id.product_uom_id._compute_quantity(
+                    mo.bom_id.product_qty, mo.product_uom_id
                 )
-                == 0
-            )
-            bypass_check = self.env.context.get("disable_check_mo_auto_validate")
-            if bypass_check:
-                return
-            if mo.bom_id and mo.auto_validate and not qty_ok:
-                raise exceptions.ValidationError(
-                    _(
-                        "The quantity to produce is restricted to {qty} "
-                        "as the BoM is configured with the "
-                        "'Order Auto Validation' option."
-                    ).format(qty=mo.bom_id.product_qty)
+
+                qty_ok = tools.float_is_zero(
```

Comment:
Why not sticking to float_compare here, like it was before?

The result won't be exactly the same.. with `float_compare`, both values are rounded and then compared.
With your new suggestion, only their diff is rounded and checked against 0.

e.g.: for the sake of a dumb example, let's assume we are rounding to 0 decimals:

value1: 2.4
value2: 1.6

before: 
- `round(value1) == round(value2)`
- `round(2.4) == round(1.6)`
- `2 == 2`

now:
- `round(value1 - value2) == 0`
- `round(2.4 - 1.6) == 0`
- `round(0.8) == 0`
- `1 == 0`